cli/src/main/java/com/liferay/blade/server/LayeredModulePathFactory.java

```diff
@@ -0,0 +1,326 @@
+
+		for (File file : modulePath) {
+
+			// Always add the root, as the user may place modules directly in it
+
+			layeredPath.add(file);
+
+			LayersConfig layersConfig = _getLayersConfig(file);
+
+			File layersDir = new File(file, layersConfig._getLayersPath());
+
+			if (FileUtil.notExists(layersDir)) {
+				if (layersConfig._isConfigured()) {
+
+					// Bad config from user
+
+					throw new IllegalStateException("No layers directory found at " + layersDir);
+				}
+
+				// else this isn't a root that has layers and add-ons
+
+				continue;
+			}
+
+			boolean validLayers = true;
+			List<File> layerFiles = new ArrayList<>();
+
+			for (String layerName : layersConfig._getLayers()) {
+				File layer = new File(layersDir, layerName);
+
+				if (FileUtil.notExists(layer)) {
+					if (layersConfig._isConfigured()) {
+
+						// Bad config from user
+
+						throw new IllegalStateException(
+							String.format("Cannot find layer %s under directory %s", layerName, layersDir));
+					}
+
+					// else this isn't a standard layers and add-ons structure
+
+					validLayers = false;
+
+					break;
+				}
+
+				loadOverlays(layer, layerFiles);
+			}
+
+			if (validLayers) {
+				foundLayers = true;
+				layeredPath.addAll(layerFiles);
+
+				// Now add-ons
+
+				File[] addOns = new File(file, layersConfig._getAddOnsPath()).listFiles();
+
+				if (addOns != null) {
+					for (File addOn : addOns) {
+						if (addOn.isDirectory()) {
+							loadOverlays(addOn, layeredPath);
+						}
+					}
+				}
+			}
+		}
+
+		if (foundLayers) {
+			return layeredPath.toArray(new File[layeredPath.size()]);
+		}
+
+		return modulePath;
+	}
+
+	public static void safeClose(Closeable closeable) {
+		if (closeable != null) {
+			try {
+				closeable.close();
+			}
+			catch (Throwable ignored) {
+			}
+		}
+	}
+
+	private static LayersConfig _getLayersConfig(File repoRoot) {
+		File layersList = new File(repoRoot, "layers.conf");
+
+		if (FileUtil.notExists(layersList)) {
+			return new LayersConfig();
+		}
+
+		Reader reader = null;
+
+		try (InputStream inputStream = Files.newInputStream(layersList.toPath())) {
+			reader = new InputStreamReader(inputStream, "UTF-8");
+
+			Properties props = new Properties();
+
+			props.load(reader);
+
+			return new LayersConfig(props);
+		}
+		catch (IOException ioe) {
+			throw new RuntimeException(ioe);
+		}
+		finally {
+			safeClose(reader);
+		}
+	}
+
+	private static final String _OVERLAYS = ".overlays";
+
+	private static class LayersConfig {
+
+		private LayersConfig() {
+			_configured = false;
+			_layersPath = _DEFAULT_LAYERS_PATH;
+			_addOnsPath = _DEFAULT_ADD_ONS_PATH;
+			_layers = Collections.singletonList("base");
+		}
+
+		private LayersConfig(Properties properties) {
+			_configured = true;
+			_layersPath = _DEFAULT_LAYERS_PATH;
+			_addOnsPath = _DEFAULT_ADD_ONS_PATH;
+
+			boolean excludeBase = false;
+
+			String layersProp = (String)properties.get("layers");
```

Comment:
This does not need to be cast to a `String`

Reply:
ok, I change it to use getProperty()